src/riak_kv_continuation.erl

```diff
@@ -0,0 +1,118 @@
+%% -------------------------------------------------------------------
+%%
+%% riak_kv_continuation: creating and decoding continuation tokens
+%%
+%% Copyright (c) 2017 Basho Technologies, Inc.  All Rights Reserved.
+%%
+%% This file is provided to you under the Apache License,
+%% Version 2.0 (the "License"); you may not use this file
+%% except in compliance with the License.  You may obtain
+%% a copy of the License at
+%%
+%%   http://www.apache.org/licenses/LICENSE-2.0
+%%
+%% Unless required by applicable law or agreed to in writing,
+%% software distributed under the License is distributed on an
+%% "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+%% KIND, either express or implied.  See the License for the
+%% specific language governing permissions and limitations
+%% under the License.
+%%
+%% -------------------------------------------------------------------
+
+%% @doc Central module for creating and decoding continuation tokens.
+
+-module(riak_kv_continuation).
+
+-export([make_token/1,
+         make_token/2,
+         make_token/3,
+         decode_token/1]).
+-export_type([token/0]).
+
+-type token() :: binary() | undefined.
+
+%% @doc Make a continuation token for the given `Term'.
+-spec make_token(term()) -> token().
+make_token(Term) ->
+    base64:encode(term_to_binary(Term)).
+
+%% @doc Make a continuation token if and only if the length of the list of
+%% `Results' is greater than `MaxResults'. If so, the item in the list at
+%% position `MaxResults' is used as the term to make the continuation token.
+%% If not, `undefined' is returned.
+%% <p>
+%% For greater control over the value used to make the continuation token, see
+%% `make_token/3', which allows for passing a function to transform the value
+%% used to make the token.
+-spec make_token(Results::list(), MaxResults::non_neg_integer()) -> token().
+make_token(Results, MaxResults) when is_list(Results),
+                                     is_integer(MaxResults),
+                                     MaxResults >= 0 ->
+    make_token(Results, MaxResults, fun(X) -> X end).
+
+%% @doc Make a continuation token if and only if the length of the list of
+%% `Results' is greater than `MaxResults'. If so, the item in the list at
+%% position `MaxResults' is used as the term to make the continuation token,
+%% after first being passed to the given `Fun' to allow for transforming the
+%% value. If not, `undefined' is returned.
```

Comment:
I've been trying to get away from this kind of use of `undefined` lately; I feel like `undefined` is overloaded to the point where its use ends up being kind of ambiguous and error-prone. If we could change it to return `'no_continuation'` or something along those lines, I think that would be much nicer.